osu.Game/Screens/OnlinePlay/Components/OnlinePlayBackgroundScreen.cs

```diff
@@ -16,65 +21,57 @@ namespace osu.Game.Screens.OnlinePlay.Components
     public abstract partial class OnlinePlayBackgroundScreen : BackgroundScreen
     {
         private CancellationTokenSource? cancellationSource;
-        private PlaylistItemBackground? background;
+        private Background? lastBackground;
+        private int? beatmapId;
 
         [BackgroundDependencyLoader]
         private void load()
         {
-            switchBackground(new PlaylistItemBackground(playlistItem));
+            loadNewBackground();
         }
 
-        private PlaylistItem? playlistItem;
-
         protected PlaylistItem? PlaylistItem
         {
-            get => playlistItem;
             set
             {
-                if (playlistItem == value)
+                if (beatmapId == value?.Beatmap.OnlineID)
                     return;
 
-                playlistItem = value;
+                beatmapId = value?.Beatmap.OnlineID;
 
-                if (LoadState > LoadState.Ready)
-                    updateBackground();
+                if (LoadState >= LoadState.Ready)
```

Comment:
Small difference here - there are no cases of `LoadState > LoadState.Ready` in the game. All others use `>=`. In other words, original code was slightly incorrect.

Though in practice this is unlikely to be hit.